mmv1/third_party/terraform/services/bigquery/resource_bigquery_table.go

```diff
@@ -862,7 +862,6 @@ func ResourceBigQueryTable() *schema.Resource {
 						"expiration_ms": {
 							Type:        schema.TypeInt,
 							Optional:    true,
-							Computed:    true,
```

Comment:
`computed` is usually set when the contributor has found that the API returns the value for the field in some scenario where the user has not set it. Is it confirmed that the API never returns a value for this when not set?

Reply:
Reached out to the API team about it and waiting for official answer there, meanwhile exploring alternatives - if I need to keep `Computed: true`, should I then set a `Default: nil` alongside it to signal "use this default value if it's not in the config"?

Also thank you for your explanation of `Computed`. Much clearer than the official documentation...

Reply:
`Computed: true` and any `Default` conflict with each other. Essentially `Optional` + `Computed` means "default to the returned API value". Also adding a new `Default` value is a breaking change.

If `Computed` is needed, I _think_ we can add custom logic to allow unsetting the value. I've seen `RawConfig` used before: https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/services/sql/resource_sql_database_instance.go.erb#L2436
In the expander function for this field, I think you can check the user value to see if it is nil, then return that instead. I'm not 100% confident, and it will need some testing however. `RawConfig` is somewhat new.

Reply:
Thank you. Meanwhile I learned why this attribute was marked `Computed: true` - BigQuery tables belong in BigQuery datasets, a separate resource, and a dataset may come with a default expiration time, which should be propagated to all the new tables created in it. In the case that a dataset default expiration is set, the absence of table-level expiration should be treated as "using the dataset-level default" instead of "no expiration".

Unless there is a way in the provider to achieve conditionally treating this value differently based on a dataset field value (is there?), given the nature of this design, I'm inclined to treat the current behavior as working-as-intended and just update documentation to suggest users to specify `expiration_ms = 0` if they need to explicitly unset. Let me know what you think.

Reply:
> Unless there is a way in the provider to achieve conditionally treating this value differently based on a dataset field value (is there?)

We have scenarios where we call the read API for one resource from within another. As far as reading the "Terraform set" value from another resource, I don't think we've ever done that (although it may be possible?)

Either way, it seems that the field should reasonably stay `Computed`.

> I'm inclined to treat the current behavior as working-as-intended and just update documentation to suggest users to specify expiration_ms = 0 if they need to explicitly unset. Let me know what you think.

That seems reasonable to me. I should have checked if `expiration_ms = 0` worked earlier, that is in line with other similar fields that are `Computed`.

Reply:
> We have scenarios where we call the read API for one resource from within another.

Thank you, this should be enough in this case, but I'll first reply in the GitHub issue explaining the situation with `Computed` and pursuing the WAI path first, unless there is strong push back there. Will comment here again once the PR is updated accordingly.

Reply:
User confirmed in https://github.com/hashicorp/terraform-provider-google/issues/15626 that leaving the field `Computed: true` and setting the value to 0 achieved the desired effect. I updated the PR to be only a documentation change (and a trivial test change).

And thank you for all the insights in this thread!

> > We have scenarios where we call the read API for one resource from within another.
> 
> Thank you, this should be enough in this case, but I'll first reply in the GitHub issue explaining the situation with `Computed` and pursuing the WAI path first, unless there is strong push back there. Will comment here again once the PR is updated accordingly.

User confirmed in https://github.com/hashicorp/terraform-provider-google/issues/15626 that leaving the field `Computed: true` and setting the value to 0 achieved the desired effect. I updated the PR to be only a documentation change (and a trivial test change).

And thank you for all the insights in this thread!